vowpalwabbit/learner.h

```diff
@@ -1063,9 +1066,9 @@ reduction_no_data_learner_builder<ExampleT, BaseLearnerT> make_no_data_reduction
   return builder;
 }
 
-template <class DataT, class ExampleT>
+template <class DataT, class BaseLearnerT, class ExampleT>
```

Comment:
Why is this needed? The point of a base learner is that it doesn't take in a base itself.

Reply:
A base learner should never actually use the base_learner parameter (pretty sure it is a nullptr here), which is why it is specified as a base_learner explicitly. I'd say this is a bug in the ftrl code and the the signature of those funcs need to change.